awswrangler/lakeformation/_utils.py

```diff
@@ -1,18 +1,134 @@
+
+
+def _get_table_objects(
+    catalog_id: Optional[str],
+    database: str,
+    table: str,
+    transaction_id: str,
+    boto3_session: Optional[boto3.Session],
+    partition_cols: Optional[List[str]] = None,
+    partitions_types: Optional[Dict[str, str]] = None,
+    partitions_values: Optional[List[str]] = None,
+) -> List[Dict[str, Any]]:
+    """Get Governed Table Objects from Lake Formation Engine."""
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+
+    scan_kwargs: Dict[str, Union[str, int]] = _catalog_id(
+        catalog_id=catalog_id,
+        **{
+            "TransactionId": transaction_id,
+            "DatabaseName": database,
+            "TableName": table,
+            "MaxResults": 100,
+        },
+    )
+    if partition_cols and partitions_types and partitions_values:
+        scan_kwargs["PartitionPredicate"] = _build_partition_predicate(
+            partition_cols=partition_cols, partitions_types=partitions_types, partitions_values=partitions_values
+        )
+
+    next_token: str = "init_token"  # Dummy token
+    table_objects: List[Dict[str, Any]] = []
+    while next_token:
+        response = client_lakeformation.get_table_objects(**scan_kwargs)
+        for objects in response["Objects"]:
+            for table_object in objects["Objects"]:
+                if objects["PartitionValues"]:
+                    table_object["PartitionValues"] = objects["PartitionValues"]
+                table_objects.append(table_object)
+        next_token = response.get("NextToken", None)
+        scan_kwargs["NextToken"] = next_token
+    return table_objects
+
+
+def _update_table_objects(
+    catalog_id: Optional[str],
+    database: str,
+    table: str,
+    transaction_id: str,
+    boto3_session: Optional[boto3.Session],
+    add_objects: Optional[List[Dict[str, Any]]] = None,
+    del_objects: Optional[List[Dict[str, Any]]] = None,
+) -> None:
+    """Register Governed Table Objects changes to Lake Formation Engine."""
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+
+    update_kwargs: Dict[str, Union[str, int, List[Dict[str, Dict[str, Any]]]]] = _catalog_id(
+        catalog_id=catalog_id,
+        **{
+            "TransactionId": transaction_id,
+            "DatabaseName": database,
+            "TableName": table,
+        },
+    )
+
+    write_operations: List[Dict[str, Dict[str, Any]]] = []
+    if add_objects:
+        write_operations.extend({"AddObject": obj} for obj in add_objects)
+    elif del_objects:
+        write_operations.extend({"DeleteObject": _without_keys(obj, ["Size"])} for obj in del_objects)
+    update_kwargs["WriteOperations"] = write_operations
+
+    client_lakeformation.update_table_objects(**update_kwargs)
+
+
```

Comment:
In LF engine, add or (mark as) deleted any passed S3 objects for a given Governed table